lib/matplotlib/colorbar.py

```diff
@@ -239,28 +239,48 @@ def __init__(self, parent, userax=True):
                 parent._axes.add_child_axes(outer_ax)
                 outer_ax._axes.child_axes.remove(parent)
             else:
-                parent.remove()
+                try:
+                    parent.remove()
+                except ValueError:
+                    pass  # Already removed
         else:
             outer_ax = parent
 
+        # swap axes in the stack if its in there:
+        if outer_ax in fig._localaxes:
+            fig._localaxes.remove(outer_ax)
+            fig._axstack.remove(outer_ax)
+            fig._localaxes.add(self)
+            fig._axstack.add(self)
         inner_ax = outer_ax.inset_axes([0, 0, 1, 1])
         self.__dict__.update(inner_ax.__dict__)
 
         self.outer_ax = outer_ax
         self.inner_ax = inner_ax
+
         self.outer_ax.xaxis.set_visible(False)
         self.outer_ax.yaxis.set_visible(False)
         self.outer_ax.set_facecolor('none')
         self.outer_ax.tick_params = self.inner_ax.tick_params
         self.outer_ax.set_xticks = self.inner_ax.set_xticks
         self.outer_ax.set_yticks = self.inner_ax.set_yticks
-        for attr in ["get_position", "set_position", "set_aspect"]:
+        for attr in ["get_position", "set_aspect",
+                     "_remove_method", "_set_position",
+                     "set_position", "cla", "draw"]:
             setattr(self, attr, getattr(self.outer_ax, attr))
         self._colorbar_info = None  # used for mpl-created axes
+        if hasattr(self.outer_ax, "get_subplotspec"):
+            attr = "get_subplotspec"
+            setattr(self, attr, getattr(self.outer_ax, attr))
+
         if userax:
             self._colorbar_info = 'user'
             # point the parent's methods all at this axes...
+            origdict = parent.__dict__
             parent.__dict__ = self.__dict__
```

Comment:
can you just update it?
```suggestion
            parent.__dict__.update(self.__dict__)
```

Reply:
Oh, maybe?  Thanks!

Reply:
No, sadly not.  I'll admit I'm probably not enough of a pythionista to dig around in the guts like this.  Its pretty apparent to me that this copies all the attributes over, but if you iterate through `__dict__` like a normal dictionary it does not capture everything in `__dict__`.  Its all a little confusing.  If someone has a more elegant less tacky way to do this, we can certainly try that approach...

Reply:
Maybe try `{**parent.__dict__, **self.__dict__}` which would produce a new dict...